src/main/ts/core/Changelog.ts

```diff
@@ -0,0 +1,361 @@
+import * as commonmark from 'commonmark';
+import { DateTime } from 'luxon';
+import * as E from 'fp-ts/Either';
+import { pipe } from 'fp-ts/pipeable';
+import * as EitherUtils from '../utils/EitherUtils';
+import { parseVersionE, Version } from './Version';
+
+interface Offset {
+  start: number;
```

Comment:
readonly on interfaces if possible.

Reply:
I've done all but ChangelogFragment. I'll need to move things around to be able to put readonly on it too so that will be another step.